esti/catalog_export_test.go

```diff
@@ -0,0 +1,418 @@
+  - id: symlink_exporter
+    type: lua
+    properties:
+      script_path: "{{ .SymlinkScriptPath }}"
+      args:
+        aws:
+          aws_access_key_id: "{{ .AccessKeyId }}"
+          aws_secret_access_key: "{{ .SecretAccessKey }}"
+          aws_region: us-east-1
+        table_source: '{{ .TableDescriptorPath }}'`
+
+func renderTplAsStr(t *testing.T, tplData any, name, content string) string {
+	t.Helper()
+	tpl, err := template.New(name).Parse(content)
+	require.NoError(t, err, "rendering template")
+	var doc bytes.Buffer
+	err = tpl.Execute(&doc, tplData)
+	require.NoError(t, err)
+	return doc.String()
+}
+
+func setupGlueClient(ctx context.Context, accessKeyID, secretAccessKey string) (*glue.Client, error) {
+	cfg, err := config.LoadDefaultConfig(ctx,
+		config.WithRegion("us-east-1"),
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(accessKeyID, secretAccessKey, "")),
+	)
+	if err != nil {
+		return nil, err
+	}
+	return glue.NewFromConfig(cfg), nil
+}
+
+func uploadAndCommitObjects(t *testing.T, ctx context.Context, repo, branch string, objectsGroups ...map[string]string) *apigen.Commit {
+	t.Helper()
+	for _, objects := range objectsGroups {
+		for path, obj := range objects {
+			resp, err := uploadContent(ctx, repo, branch, path, obj)
+			require.NoError(t, err)
+			require.Equal(t, http.StatusCreated, resp.StatusCode())
+		}
+	}
+
+	commitResp, err := client.CommitWithResponse(ctx, repo, mainBranch, &apigen.CommitParams{}, apigen.CommitJSONRequestBody{
+		Message: "Table Data",
+	})
+	require.NoErrorf(t, err, "failed commiting uploaded objects to lakefs://%s/%s", repo, branch)
+	require.Equal(t, http.StatusCreated, commitResp.StatusCode())
+	return commitResp.JSON201
+}
+func genCSVData(cols []string, n int) string {
+	csvBlob := ""
+	// generate header
+	for _, c := range cols {
+		csvBlob += c + ","
+	}
+	csvBlob = strings.TrimSuffix(csvBlob, ",") + "\n"
+	// generate rows
+	for rowNum := 0; rowNum < n; rowNum++ {
+		row := ""
+		for i := range cols {
+			row += fmt.Sprintf("%d,", rowNum+i)
+		}
+		csvBlob += strings.TrimSuffix(row, ",") + "\n"
+	}
+	return csvBlob
+}
+
+func testSymlinkS3Exporter(t *testing.T, ctx context.Context, repo string, tablePaths map[string]string, testData *exportHooksTestData) (string, string) {
+	t.Helper()
+
+	tableYaml, err := yaml.Marshal(&testData.TableSpec)
+
+	if err != nil {
+		require.NoError(t, err, "failed marshaling table spec to YAML")
+	}
+
+	hookFiles := map[string]string{
+		testData.SymlinkScriptPath:   symlinkExporterScript,
+		testData.TableDescriptorPath: renderTplAsStr(t, testData.TableSpec, "table", string(tableYaml)),
+	}
+
+	// upload table objects and action script
+	uploadAndCommitObjects(t, ctx, repo, mainBranch, hookFiles, tablePaths)
+
+	// upload action
+	commit := uploadAndCommitObjects(t, ctx, repo, mainBranch, map[string]string{
+		"_lakefs_actions/animals_symlink.yaml": renderTplAsStr(t, testData, "action", symlinkExporterAction),
```

Comment:
Why does this upload need to be separate from the one on l.196?

Reply:
I don't you're right consolidated everything into one upload, thanks!